src/LiquidationBot.ts

```diff
@@ -254,24 +257,32 @@ export default class LiquidationBot {
 
   async run() {
     const users = await this.computeLiquidableUsers();
-    const liquidationsParams = await Promise.all(
-      users.map((u) => this.getUserLiquidationParams(u.address))
-    );
-    const toLiquidate = (
-      await Promise.all(
-        liquidationsParams.map(async (user) => {
-          if (
-            await this.isProfitable(
-              user.debtMarket.market,
-              user.toLiquidate,
-              user.debtMarket.price
+    this.logger.log(`Found ${users.length} users liquidatable`);
+    // use the batch size to limit the number of users to liquidate
+    const toLiquidate: UserLiquidationParams[] = [];
+    for (let i = 0; i < users.length; i += this.settings.batchSize) {
```

Comment:
isn't it missing a random delay or a retry strategy?

Reply:
this is another issue yes, I'm thinking about using rxjs for that